.github/workflows/python-package.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
```

Comment:
can we add 3.9, 3.10, 3.11, 3.12?

Reply:
3.12 failed due to the removal of `distutils` package https://stackoverflow.com/questions/77247893/modulenotfounderror-no-module-named-distutils-in-python-3-12

other versions on the way

Reply:
3.10 and 3.11 also failed. we need more time to get it pass.
I noticed scanpy is asking a python version <3.10, we may remain in 3.9 temporarily.
https://github.com/scverse/scanpy/blob/db2118e8eb60bbf6287ce1413477480e16b2508b/pyproject.toml#L70